### PR TITLE
added tasks related "exceptions"

### PR DESCRIPTION
**Pull request by Ypostolov**

### CODE REVIEW

File: week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  Rational result;
+  result.Initialize((right.Numerator() * left.Numerator()),
+                    (right.Denominator() * left.Denominator()));
+  return result;
+}
+Rational operator/(const Rational& left, const Rational& right) {
+  if (right.Numerator() == 0) {
+    throw std::domain_error("WTF?");
+  } else {
+    Rational result;
```
Comment (ZhdanovP) [inline week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp:133]: @Ypostolov let's do this operators as a part of the class

**Review by ZhdanovP — CHANGES_REQUESTED** [review]

File: week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp (hunk 1 above)
Comment (Ypostolov) [reply week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp:133]: Implemented in 46ae2ab8230721ce3aff84a15e7f276e7f8e892d.
Please take a look.

File: week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp [hunk 2]
```diff
@@ -116,39 +116,40 @@ class Rational {
     return result;
   }
 
- private:
-  int p;
-  int q;
-};
+  friend std::istream& operator>>(std::istream& stream, Rational& r) {
```
Comment (ZhdanovP) [inline week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp:119]: @Ypostolov Why have you made this is a friend function, but not class member-method?

Comment (Ypostolov) [reply week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp:119]: Fixed in 6837310212b92d0b5ad8047ce10bcd625a6c1f84
Please take a look.

**Pull request merged**